Add nixkell to the Haskell subsection (#84)
nix-community/awesome-nix: README.md

* [stack2nix](https://github.com/input-output-hk/stack2nix) - Generate nix expressions for Haskell projects.
* [nix-haskell-mode](https://github.com/matthewbauer/nix-haskell-mode) - Automatic Haskell setup in Emacs.
* [haskell.nix](https://github.com/input-output-hk/haskell.nix) - Alternative Haskell Infrastructure for Nixpkgs.

> * [nixkell](https://github.com/pwm/nixkell) - Get your Haskell projects up and running with no fuss using Nix.

Reviewer: It took me a while to figure out what the project is doing.
How do you think about a project description like this?

```suggestion
* [nixkell](https://github.com/pwm/nixkell) - A Haskell project template using Nix and direnv.
```
